website/api/controllers/try-fleet/view-query-report.js

```diff
@@ -0,0 +1,237 @@
+    },
+
+  },
+
+
+  fn: async function ({hostPlatform, tableName}) {
+
+    if(!_.isObject(sails.config.builtStaticContent) || !_.isArray(sails.config.builtStaticContent.osqueryTables)){
+      throw {badConfig: 'builtStaticContent.osqueryTables'};
+    }
+
+    // If the requesting user is not logged in, redirect them to the /try-fleet/login page with the specified hostPlatform added as a query parameter.
+    if(!this.req.me){
+      throw {redirect: `/try-fleet/login?targetPlatform=${hostPlatform}` };
+    }
+
+    if(!sails.config.custom.queryIdsByTableName){
+      throw new Error('Missing config variable: The dictionary of query ids required to use the query-report page is missing! (sails.config.custom.queryIdsByTableName)');
+    }
+
+    if(!sails.config.custom.hostIdsByHostPlatform){
+      throw new Error('Missing config variable: The dictionary of host ids required to use the query-report page is missing! (sails.config.custom.hostIdsByHostPlatform)');
+    }
+
+    if(!sails.config.custom.teamApidForQueryReports){
+      throw new Error('Missing config variable: The id of the team the query report page gets results for is missing! (sails.config.custom.teamApidForQueryReports)');
+    }
+
+    if(!sails.config.custom.fleetBaseUrlForQueryReports){
+      throw new Error('Missing config variable: The URL of the fleet instance used for query reports is missing! (sails.config.custom.fleetBaseUrlForQueryReports)');
+    }
+
+    if(!sails.config.custom.fleetTokenForQueryReports){
+      throw new Error('Missing config variable: The API token for requests to the Fleet instance used for queyr reports is missing! (sails.config.custom.fleetTokenForQueryReports)');
+    }
+
+    //  ┬ ┬┌─┐┌─┐┌┬┐  ┬┌┐┌┌─┐┌─┐┬─┐┌┬┐┌─┐┌┬┐┬┌─┐┌┐┌
+    //  ├─┤│ │└─┐ │   ││││├┤ │ │├┬┘│││├─┤ │ ││ ││││
+    //  ┴ ┴└─┘└─┘ ┴   ┴┘└┘└  └─┘┴└─┴ ┴┴ ┴ ┴ ┴└─┘┘└┘
+    let hostIdsByHostPlatform = sails.config.custom.hostIdsByHostPlatform;
+    // Get the ID of the host we'll be showing results for.
+    let selectedHostId = hostIdsByHostPlatform[hostPlatform];
+
+    // Send an HTTP request to get the host details for hosts on the query report team.
+    let hostsOnQueryReportTeamApiResponse = await sails.helpers.http.get.with({
+      url: sails.config.custom.fleetBaseUrlForQueryReports+'/api/v1/fleet/hosts?team_id='+sails.config.custom.teamApidForQueryReports,
+      headers: {
+        Authorization: `Bearer ${sails.config.custom.fleetTokenForQueryReports}`
+      }
+    })
+    .intercept((error)=>{
+      return new Error(`When sending an API request to ${sails.config.custom.fleetBaseUrlForQueryReports}/api/v1/fleet/hosts?team_id=${sails.config.custom.teamApidForQueryReports} to get information about hosts on the query report team, an error occured: ${error}`);
+    });
+    if(!hostsOnQueryReportTeamApiResponse.hosts || hostsOnQueryReportTeamApiResponse.hosts.legnth < 1){
+      // TODO: is there a better way of handling this error?
```

Comment:
when would it happen?

Reply:
when the API sends back malformed/unexpected response?

that's part of it.  In that case it's like a consistency violation

The other part is if the hosts have been unenrolled.

I'd expect a 5xx from both, but different error message from each.

The erorr message here is for the latter case